### DISCUSSION TITLE
Logging of failed login into UI

### DISCUSSION THREAD

**Discussion by 999LV** [zwave-js/zwave-js-ui#2933] [Q&A]
I would like to understand if/how invalid UI logins are logged, for the purpose of using fail2ban to deny brute force attacks on the UI. Running version 8.8.0 as a package (not in a container) on PiOS Bullseye, I enabled logging to file in the UI / Settings section with logging level set to « warn » but I cannot see any logging…
Thanks in advance !

**Comment by robertsLando** [comment]
Created an issue for this #2934

**Comment by 999LV** [comment]
@robertsLando, Thanks for making this a proposed enhancement… I am just a newbie with zwave-js-ui  (and cannot help with Node JS, unlike python or c) but I figure that to achieve what I was asking for, the UI logging needs to be tweaked to have a persistent file name (could be a link similar to the zwavejs_current.log?) for fail2ban to monitor. The current working of z-ui_YYYY-MM-DD.log does not lend itself to such monitoring.
Not sure if you are familiar with fail2ban but obviously to allow banning, the log entry of the failed login needs to include the source IP (either ipv4 or ipv6 as appropriate).
Many thanks !

**Reply by robertsLando** [reply]
Done

**Reply by robertsLando** [reply]
FYI there is a rate limiter on ui that automatically handles such things

**Reply by 999LV** [reply]
Many thanks. Will definitely give it a try !
